autorest/azure/async.go

```diff
@@ -434,16 +464,16 @@ func updatePollingState(resp *http.Response, ps *pollingState) error {
 	if ps.hasFailed() {
 		if ps.PollingMethod == PollingAsyncOperation {
 			or := pt.(*operationResource)
-			ps.Code = or.OperationError.Code
-			ps.Message = or.OperationError.Message
+			ps.ServiceError = &or.OperationError
 		} else {
 			p := pt.(*provisioningStatus)
 			if p.hasProvisioningError() {
-				ps.Code = p.ProvisioningError.Code
-				ps.Message = p.ProvisioningError.Message
+				ps.ServiceError = &p.ProvisioningError
 			} else {
```

Comment:
is this for  "just in case" it happens ? otherwise when would it be an `unknown` case ?

Reply:
Yes I believe this is a fall-back for a "just in case" (it's been like this since April 2016).  That said, hitting this clause is indicative of a bug somewhere in the system, i.e. either the service didn't return any error info, it was in a format that we couldn't unmarshal, or some other bug in our runtime.